reana-ui/src/pages/launchOnReana/LaunchOnReana.js

```diff
@@ -76,37 +87,107 @@ export default function LaunchOnReana() {
     return !requiredParams.every((param) => qsKeys.includes(param));
   }
 
+  /**
+   * Memoised function to get workflow parameters.
+   * Since this function is called twice and its return value should not
+   * change unless the query string parameters change, memoising it improves
+   * performace, avoiding extra calls.
+   */
+  const getWorkflowParameters = useCallback(() => {
+    /**
+     * Parses workflow parameters. Triggers notification error if invalid.
+     * @param {URLSearchParams} query Standard URLSearchParams object.
+     * @returns {Object | null} Returns object with workflow parameters or `null` if invalid.
+     */
+    function fn(query) {
+      try {
+        return JSON.parse(query.get("parameters"));
+      } catch (error) {
+        dispatch(
+          triggerNotification(
+            "An error has occurred",
+            `Invalid JSON workflow parameters provided: ${query.get(
+              "parameters"
+            )}`,
+            { error: true }
+          )
+        );
+      }
+      return null;
+    }
+    return fn(query);
+  }, [dispatch, query]);
+
   return (
     <BasePage>
       <Container text className={styles["container"]}>
         <div>
           <Title>Launch on REANA</Title>
         </div>
         {isMissingRequiredParams(query) ? (
-          <Message
-            icon="warning circle"
-            header="Missing required query string fields."
-            content={`The required fields: ${getRequiredQsParams()}`}
-            warning
-          />
+          <Welcome />
         ) : (
-          <ul>
-            {[...query].map(([k, v]) => (
-              <li key={k}>
-                {k}: {v}
-              </li>
-            ))}
-          </ul>
+          <Box className={styles.box} wrap>
+            <section className={styles.details}>
+              <span className={styles.icon}>
+                <Icon name="hourglass half" />
+              </span>
+              <div>
+                <div className={styles.name}>
+                  {query.get("name") ?? DEFAULT_WORKFLOW_NAME}
+                </div>
+                <a className={styles.url} href={query.get("url")}>
+                  {query.get("url")}
```

Comment:
Very minor: I personally get a feeling that `name`, `url` and `spec` are too close to each other? This is perhaps to replicate the same look and feel as in workflow list page, but maybe we could increase the space between these elements just by a little bit, WDYT? I understand that it's a matter of taste, so please ignore my suggestions if you don't agree

Reply:
Applied 5px padding.